### PR TITLE
[Core] Adding tensor variable by components definition

### LINKED ISSUE #2182: [Core] Registering a variable with custom number of components?
https://github.com/KratosMultiphysics/Kratos/blob/4098f560ca06ba74977ae199cd59207aa98e7334/kratos/includes/define.h#L126

Can we define a macro to create variable with custom number of components?, @marandra is interested in this feature in order to avoid register 6 different LM variables

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #2182

Propossal, any suggested names?

### CODE REVIEW

**Comment by loumalouomega** [comment]
I was thinking that maybe for non-symmetric tensor is to use a BoundedMatrix, but that would need enhance the adaptor

**Review by pooyan-dadvand — COMMENTED** [review]
I agree with the implementation

However someone else should check the order you have used in tensors.

**Comment by loumalouomega** [comment]
Is the order used in the math_utils https://github.com/KratosMultiphysics/Kratos/blob/2e72c5fdd9445ac996eba9e3b3cf8ec3d8020296/kratos/utilities/math_utils.h#L1150

**Comment by pooyan-dadvand** [comment]
@RiccardoRossi ?

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
@RiccardoRossi

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
This is ready to  merge

**Comment by loumalouomega** [comment]
Ping @RiccardoRossi

**Comment by loumalouomega** [comment]
Ping

File: kratos/includes/define_python.h [hunk 1]
```diff
@@ -35,6 +35,52 @@
...
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(name##_YY) \
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(name##_ZZ) \
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(name##_XY) \
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(name##_YZ) \
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(name##_XZ)
+
+#ifdef KRATOS_REGISTER_IN_PYTHON_2D_TENSOR_VARIABLE_WITH_COMPONENTS
+#undef KRATOS_REGISTER_IN_PYTHON_2D_TENSOR_VARIABLE_WITH_COMPONENTS
+#endif
+#define KRATOS_REGISTER_IN_PYTHON_2D_TENSOR_VARIABLE_WITH_COMPONENTS(name) \
```
Comment (RiccardoRossi) [inline kratos/includes/define_python.h:62]: wouldn't it be better to use the NON_SYMMETRIC keyword to make it very explicit in the macro name?

**Review by RiccardoRossi — APPROVED** [review]

File: kratos/includes/define_python.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/includes/define_python.h:62]: Humm, could be, in any case it is a convection name

**Pull request merged**